### PR TITLE
Add implementation: ws_value type

### PR DESCRIPTION
**Pull request by matthiasbeyer**
Targets #34

### CODE REVIEW

File: src/values/value.h [hunk 1]
```diff
@@ -28,6 +28,44 @@
...
 
+typedef void (ws_value_type_deinit_callback)(struct ws_value*);
+
+struct ws_value {
+    enum ws_value_type type;
+    ws_value_type_deinit_callback deinit_callback;
+};
+
+/**
+ * Initialize an value object
```
Comment (xVin) [inline src/values/value.h]: "an" --> "a"

File: src/values/value.h [hunk 2]
```diff
@@ -28,6 +28,44 @@
...
+ */
+void
+ws_value_init(
+    struct ws_value* self //!< The value object to initialize
+)
+__ws_nonnull__(1)
+;
+
+/**
+ * Deinit an value object
```
Comment (xVin) [inline src/values/value.h]: "an" --> "a"

File: src/values/value.h [hunk 3]
```diff
@@ -28,6 +28,44 @@
...
+ */
+void
+ws_value_deinit(
+    struct ws_value* self //!< The value object to deinitialize
+)
+__ws_nonnull__(1)
+;
+
+/**
+ * Get the type of an ws_value object
```
Comment (xVin) [inline src/values/value.h]: "an" --> "a"

File: src/values/value_type.h [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+enum ws_value_type {
+    WS_VALUE_TYPE_NONE,     //!< Indicator: This is not a value object anymore
+    WS_VALUE_TYPE_VALUE,
+    WS_VALUE_TYPE_NIL,
+    WS_VALUE_TYPE_BOOL,
+    WS_VALUE_TYPE_STRING,
+    WS_VALUE_TYPE_INT,
+    WS_VALUE_TYPE_OBJECT_ID,
+    WS_VALUE_TYPE_SET,
+    WS_VALUE_TYPE_NAMES,
```
Comment (neithernut) [inline src/values/value_type.h]: Should be `WS_VALUE_TYPE_NAMED`.

File: src/values/value_type.h [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+/**
+ * Type identifiers for value objects
+ */
+enum ws_value_type {
+    WS_VALUE_TYPE_NONE,     //!< Indicator: This is not a value object anymore
+    WS_VALUE_TYPE_VALUE,
+    WS_VALUE_TYPE_NIL,
+    WS_VALUE_TYPE_BOOL,
+    WS_VALUE_TYPE_STRING,
+    WS_VALUE_TYPE_INT,
```
Comment (neithernut) [inline src/values/value_type.h:39]: Maybe put int before string.

**Comment by neithernut** [comment]
GO for squashing.

**Comment by matthiasbeyer** [comment]
No, thanks.

**Comment by neithernut** [comment]
Then ACK.

**Pull request merged**